string_manipulation.rb

```diff
@@ -13,10 +13,29 @@ def string_reverse(my_string)
   return reversed_string
 end
 
-
+# puts "Test 2: reversed words"
+# my_words = "I can be an  engineer"
+# puts "Original: #{my_words}"
+# reversed_words = "I nac eb na  reenigne"
+# reverse_words(my_words)
 # aA method to reverse each word in a sentence, in place.
 def reverse_words(my_words)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  start = 0
+  segment = ""
+
+  while i < my_words.length
+    if my_words[i] == " "
+      my_words[start..(i-1)] = string_reverse(segment)
+      segment = ""
+      start = i + 1
+    else
+      segment << my_words[i]

```

Comment:
You are creating new memory here by adding each character in every word to segment. So the current space complexity of your method is O(n) if there is only one word in my_words or O(k) where k is the length of the longest word in the input string. Can you author this method without using this additional space and indeed make your space complexity O(1)?